fast_llm/functional/cross_entropy.py

```diff
@@ -41,6 +51,10 @@ def fused_cross_entropy_forward_backward(
     """
     # Do the forward and backward passes all at once, and fused with dtype conversion.
     # Way faster and more memory-efficient than the pytorch version.
+    if apply_loss_mask:
+        loss_mask = target != ignore_index
```

Comment:
why are we first integrating the loss mask with the target tensor and then extract it here again? does this save any noteworthy amount of memory? why not passing an explicit and optional binary loss mask to the cross entropy function?

Reply:
cc @jlamypoirier

Reply:
I don't want to delay this PR unnecessarily. If you and @jlamypoirier think that the combo of ignore indices in the `target` and `apply_loss_mask` is the right interface for the cross entropy loss, then let's leave it like that.